site/layouts/partials/course_info.html

```diff
@@ -47,9 +47,13 @@ <h5 class="font-weight-bold">Course Info</h5>
         <td class="pl-0 pb-3">Topic(s)</td>
         <td class="pb-3">
           <ul class="list-unstyled m-0">
-            {{ range $topic := .Params.course_info.topics }}
+            {{ range $topicObject := .Params.course_info.topics }}
             <li>
               <a href="#" class="coming-soon">
+                {{- $topic := "" -}}
+                {{- $topic := printf "%s%s" $topic (cond (isset $topicObject "topic") $topicObject.topic "") -}}
+                {{- $topic := printf "%s%s" $topic (cond (isset $topicObject "subtopic") (printf " > %s" $topicObject.subtopic) "") -}}
+                {{- $topic := printf "%s%s" $topic (cond (isset $topicObject "speciality") (printf " > %s" $topicObject.speciality) "") -}}
```

Comment:
this is a bit silly but I was thinking there must be a cleaner way to do this - but hugo is kind of inflexible with certain things! like if you can an array that looks like `["string", "string", nil]` there doesn't seem to be a good way to get rid of the `nil` - `where` seems to only work with objects that have a parameter! Agh.

but you can use the `complement` function to remove items from a collection, then you can use the `delimit` function to intersperse elements into a slice. so if you wanted I think this would work:

```
{{- $topics := slice $topicObject.topic $topicObject.subtopic $topicObject.speciality -}}
{{ delimit ($topics | complement (slice nil )) " > " }}
```

Reply:
Neat!  I pushed this up because it works and is cleaner.  We will be changing how this is rendered once @abdulkdawson is done with the new "pill" design for it, so really the goal in this PR is just to be able to at least render the topics using the new structure.